src/main/java/net/minestom/server/item/meta/MapMeta.java

```diff
@@ -0,0 +1,278 @@
+package net.minestom.server.item.meta;
+
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.chat.ChatColor;
+import net.minestom.server.color.Color;
+import net.minestom.server.item.ItemMeta;
+import net.minestom.server.item.ItemMetaBuilder;
+import net.minestom.server.utils.clone.PublicCloneable;
+import org.jetbrains.annotations.NotNull;
+import org.jglrxavpok.hephaistos.nbt.NBTCompound;
+import org.jglrxavpok.hephaistos.nbt.NBTList;
+import org.jglrxavpok.hephaistos.nbt.NBTTypes;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Supplier;
+
+public class MapMeta extends ItemMeta {
+
+    private final int mapId;
+    private final int mapScaleDirection;
+    private final List<MapDecoration> decorations;
+    private final Color mapColor;
+
+    protected MapMeta(ItemMetaBuilder metaBuilder,
+                      int mapId,
+                      int mapScaleDirection,
+                      @NotNull List<MapDecoration> decorations,
+                      @NotNull Color mapColor) {
+        super(metaBuilder);
+        this.mapId = mapId;
+        this.mapScaleDirection = mapScaleDirection;
+        this.decorations = decorations;
+        this.mapColor = mapColor;
+    }
+
+    /**
+     * Gets the map id.
+     *
+     * @return the map id
+     */
+    public int getMapId() {
+        return mapId;
+    }
+
+    /**
+     * Gets the map scale direction.
+     *
+     * @return the map scale direction
+     */
+    public int getMapScaleDirection() {
+        return mapScaleDirection;
+    }
+
+    /**
+     * Gets the map decorations.
+     *
+     * @return a modifiable list containing all the map decorations
+     */
+    public List<MapDecoration> getDecorations() {
```

Comment:
`@Nullable`?

Reply:
Same reasoning as for the lore, should we perhaps consider all lists non-null but possibly empty?

Reply:
Empty lists is definitely better than nullable lists imo